### DISCUSSION TITLE
Role Permissions Mangement

### DISCUSSION THREAD

**Discussion by EricBui619** [OrchardCMS/OrchardCore#9082] [Q&A] [answered]
How can I add a new permission, edit definition of an exist permission and delete it if I want for a specified role? I've tried to use IPermissionProvider but it juts runs only one time because when I modified it, nothing happens. So, what I need to do to trigger it again or any way more flexible that I can use to modify my custom permissions for a module?

**Comment by deanmarcussen** [comment, marked as answer]
The `RoleUpdate` runs when features are installed, so when you are developing new permissions, you need to disable and reenable your feature
